sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -3957,13 +3957,16 @@ object ApplyCharTypePadding extends Rule[LogicalPlan] {
           if attr.dataType == StringType && list.forall(_.foldable) =>
           CharVarcharUtils.getRawType(attr.metadata).flatMap {
             case CharType(length) =>
-              val literalCharLengths = list.map(_.eval().asInstanceOf[UTF8String].numChars())
-              val targetLen = (length +: literalCharLengths).max
-              Some(i.copy(
-                value = addPadding(attr, length, targetLen),
-                list = list.zip(literalCharLengths).map {
-                  case (lit, charLength) => addPadding(lit, charLength, targetLen)
-                }))
+              val (nulls, literalChars) =
+                list.map(_.eval().asInstanceOf[UTF8String]).partition(_ == null)
+              val literalCharLengths = literalChars.map(_.numChars())
+              val targetLen = literalCharLengths.fold(length)(_ max _)
+
+              val newAttr = addPadding(attr, length, targetLen)
+              val newList = list.zip(literalCharLengths).map {
+                case (lit, charLength) => addPadding(lit, charLength, targetLen)
+              } ++ nulls.map(Literal.create(_, StringType))
+              Some(i.copy(value = newAttr, list = newList))
```

Comment:
If you don't mind, shall we keep the change minimal? For example, it seems that we don't need to change the following lines.
```scala
Some(i.copy(
  value = addPadding(attr, length, targetLen),
```

Reply:
+1, let's minimize the diff

Reply:
OK, got it